tests/query/test_query.py

```diff
@@ -154,9 +154,18 @@ def test_batches_strip_comments(self):
         expected_selections = [
             SelectionData(start_line=0, start_column=0, end_line=0, end_column=17),
             SelectionData(start_line=3, start_column=1, end_line=3, end_column=18)]
+
         for index, batch in enumerate(query.batches):
             self.assertEqual(_tuple_from_selection_data(batch.selection), _tuple_from_selection_data(expected_selections[index]))
 
+    def test_hash_character_processed_correctly(self):
+        """Test that xor operator is not taken for an inline comment delimiter"""
+        full_query = "select 42 # 24;"
+        query = Query('test_uri', full_query, QueryExecutionSettings(ExecutionPlanOptions(), None), QueryEvents())
+
+        self.assertEqual(len(query.batches), 1)
+        self.assertEqual(full_query, query.batches[0].batch_text)
+
```

Comment:
Is it possible to test the evaluation of the query? You mentioned that "select 42 # 24;" should evaluate to 50. Another assertion to check that could be good to make sure its evaluating correctly.

Reply:
Yes, it is possible, but it would require writing an integration test. This specific test verifies that the query text is not getting damaged when it is wrapped into the query object. And if it is not than we can trust that the server will evaluate it correctly.